codex/codex.nim

```diff
@@ -123,17 +124,25 @@ proc new(_: type Contracts,
     error "Marketplace contract address not found in deployment file"
     quit QuitFailure
 
-  # TODO: at some point there may be cli options that enable client-only or host-only
-  # operation, and both client AND host will not necessarily need to be instantiated
-  let client = ClientInteractions.new(config.ethProvider,
-                                      account,
-                                      marketplaceAddress)
-  let host = HostInteractions.new(config.ethProvider,
-                                  account,
-                                  repo,
-                                  marketplaceAddress)
-
-  (client.option, host.option)
+  let provider = JsonRpcProvider.new(config.ethProvider)
+  let signer = provider.getSigner(account)
+  let marketplace = Marketplace.new(marketplaceAddress, signer)
+  let market = OnChainMarket.new(marketplace)
+  let clock = OnChainClock.new(provider)
+  let purchasing = Purchasing.new(market, clock)
```

Comment:
In the future, when we may not necessarily want to run a client, the `purchasing` module will not need to be instantiated. The same is true for the `proving` module if we're in client-only mode. This is one of the reasons I kept the construction of these modules inside of `ClientInteractions.new` and `HostInteractions.new`.